docs/tests/add-new-PR-checkers.md

```diff
@@ -0,0 +1,120 @@
+# Steps to add a New PR Checker for Impacted Area-Based PR Testing
+
+### Background
+In our PR testing, each PR checker represents a specific topology and runs the corresponding test cases. As more topology types are added to production environments, we need to add additional PR checkers. This document will guide you through the process of adding a new PR checker in impacted area-based PR testing.
+
+### Location of PR Checkers
+PR checkers are defined in the `sonic-mgmt/azure-pipelines.yml` file. Each job represents a PR checker. Here’s a template example of the job:
+```
+- job: impacted_area_dualtor_elastictest  
+  displayName: "impacted-area-kvmtest-dualtor by Elastictest"  
+  dependsOn:  
+  - get_impacted_area  
+  - choose_between_mixed_and_py3_ptf_image  
+  condition: contains(dependencies.get_impacted_area.outputs['SetVariableTask.PR_CHECKERS'], 'dualtor_checker')  
+  variables:  
+    TEST_SCRIPTS: $[ dependencies.get_impacted_area.outputs['SetVariableTask.TEST_SCRIPTS'] ]  
+    set_ptf_image_tag: $[ dependencies.choose_between_mixed_and_py3_ptf_image.outputs['ptf_image_tag.tag_value'] ]  
+  timeoutInMinutes: 240  
+  continueOnError: false  
+  pool: sonic-ubuntu-1c  
+  steps:  
+    - template: .azure-pipelines/impacted_area_testing/calculate-instance-numbers.yml  
+      parameters:  
+        TOPOLOGY: dualtor  
+        BUILD_BRANCH: $(BUILD_BRANCH)  
+        # 30 mins for preparing testbed, 30 mins for pre-test and 20 mins for post-test  
+        PREPARE_TIME: 80  
+  
+    - template: .azure-pipelines/run-test-elastictest-template.yml  
+      parameters:  
+        TOPOLOGY: dualtor  
+        PTF_IMAGE_TAG: $(set_ptf_image_tag)  
+        SCRIPTS: $(SCRIPTS)  
+        MIN_WORKER: $(INSTANCE_NUMBER)  
+        MAX_WORKER: $(INSTANCE_NUMBER)  
+        COMMON_EXTRA_PARAMS: "--disable_loganalyzer "  
+        KVM_IMAGE_BRANCH: $(BUILD_BRANCH)  
+        MGMT_BRANCH: $(BUILD_BRANCH)
+```
+
+### Steps to Add a New PR Checker
+**1. Update Test Scripts with Topology Marks**
+
+Each test script includes a pytest mark to specify supported topologies. For your new PR checker, ensure the relevant test scripts include the new topology mark:
+
+Example: 
```

Comment:
Suggest giving concrete examples to guide people setting up new PR checks.

Reply:
Add a template in pytestmark